types/usb/index.d.ts

```diff
@@ -89,41 +93,41 @@ export class Device {
    * @param desc_index
    * @param callback
    */
-  getStringDescriptor(desc_index: number, callback: (error?: string, buf?: Buffer) => void): void;
+  getStringDescriptor(desc_index: number, callback: (error?: LibUSBException, data?: string) => void): void;
```

Comment:
This is a big type change, concerned the error type change may cause issues, but the Buffer to string change certainly will.

Reply:
But `getStringDescriptor` has [always returned a string](https://github.com/tessel/node-usb/blame/a69f1e9cb83d11d88cb009013b7346f8c3eac3a4/usb.js#L160), the typings were just incorrect.
If this change causes issues in someone's code, it means the code was invalid to begin with.

Reply:
I agree and your change is correct. Just highlighting this change can break type checking for current users.

Reply:
I wonder if we can update the typings version to be a major change and highlight this could break TS compiles?

Reply:
AFAIK, projects won't be broken until they manually update (due to package lock).
In any case, I'm new to DefinitelyTyped and don't know how typings are versioned, or how these kind of breaking changes are managed, so I'll be happy with whatever approach you see more convenient :)